src/api/cocktaildb.ts

```diff
@@ -47,14 +47,29 @@ async function searchByIngredient(ingredient: string): Promise<Drink[]> {
   }
 }
 
-async function searchByMultipleIngredients(
+export async function searchByMultipleIngredients(
   ingredients: string[],
 ): Promise<Drink[]> {
   try {
     const responses = await Promise.all(
       ingredients.map((ingredient) => searchByIngredient(ingredient)),
     );
-    return responses.flat();
+    const drinkIdsInResponses = [];
+    for (let i = 0; i < ingredients.length; i++) {
+      drinkIdsInResponses[i] = responses[i].map((drink) => drink.idDrink);
+    }
+
+    const intersection: string[] = [];
```

Comment:
This is a great approach, @cesine 🙌🏾 

I see use cases when users want to match all the ingredients in their generated cocktail. There might also be other use cases where users could list the available ingredients and see what will be a recommended cocktail taking at least one ingredient available. 

I wonder if we could offer both options, for instance having an explicit checkbox saying "Include all ingredients to generate cocktail" (maybe a better copy 😅 )

Reply:
sounds good! i will update the PR

Reply:
I updated the PR to remove this logic and instead make it feel like you have a well stocked cabinet by default https://github.com/axelavargas/ai-buddy-cocktail-gen-code/pull/2#discussion_r1885013929